swagger_server/utils/db_utils.py

```diff
@@ -1,67 +1,51 @@
+        self.mongo_client = pymongo.MongoClient(MONGODB_CONNSTRING)
         self.logger = logging.getLogger(__name__)
-        # self.logger.setLevel(logging.DEBUG)
+        self.logger.setLevel(logging.DEBUG)
 
-    def _initialize_db(self, db_filename, db_tables_tuples, print_table_on_load=False):
-        # DB related utils
-        # Details on the setup:
-        # https://dataset.readthedocs.io/en/latest/api.html
-        # https://github.com/g2p/bedup/issues/38#issuecomment-43703630
-        self.logger.debug("Connection to DB: %s" % db_filename)
-        self.db = dataset.connect(
-            "sqlite:///" + db_filename,
-            engine_kwargs={"connect_args": {"check_same_thread": False}},
-        )
-
-        # Load the tables, create table if table does not.
-        for name, table in db_tables_tuples:
-            self.logger.debug("DB name: {}, table name: {}".format(name, table))
-            if table in self.db:  # https://github.com/pudo/dataset/issues/281
-                self.logger.debug("Trying to load {} from DB".format(name))
-                t = self.db.load_table(table)
-                if print_table_on_load:
-                    entries = t.find()
-                    self.logger.debug("\n\n&&&&& ENTRIES in %s &&&&&" % name)
-                    for e in entries:
-                        self.logger.debug("\n%s" % str(e))
-                    self.logger.debug("&&&&& END ENTRIES &&&&&\n\n")
-                setattr(self, name, t)
+    def _initialize_db(self):
+        self.logger.debug("Trying to load {} from DB".format(self.db_name))
 
-            else:
-                # If load_table() fails, that's fine! It means that the
-                # table doesn't yet exist. So, create it.
-                self.logger.debug("No existing {} from DB, creating table".format(name))
-                t = self.db[table]
-                setattr(self, name, t)
+        if self.db_name not in self.mongo_client.list_database_names():
+            self.logger.debug(
+                "No existing {} from DB, creating table".format(self.db_name)
+            )
+            self.sdxdb = self.mongo_client[self.db_name]
+            self.logger.debug("DB {} initialized".format(self.db_name))
 
-    def read_from_db(self, key):
-        # Returns the manifest filename if it exists or None if it does not.
-        d = self.config_table.find_one(key=key)
-        if d == None:
-            self.logger.debug("Did not find entry.")
-            return None
-        value = d["value"]
-        self.logger.debug("DB return value: " + str(value))
-        return value
+        self.sdxdb = self.mongo_client[self.db_name]
+        config_col = self.sdxdb[self.config_table_name]
+        self.logger.debug("DB {} initialized".format(self.db_name))
 
     def add_key_value_pair_to_db(self, key, value):
-        # Pushes key-value to DB.
-        if self.read_from_db(key) == None:
+        obj = self.read_from_db(key)
```

Comment:
Some error handling might be useful here: what happens if `key` or `value` is invalid?  I guess `key` has to be a string, and `value` has to be JSON?  Should we check input, or just let the method crash?

Reply:
We need some tests for `DbUtils` class. :-)

Reply:
If the key is invalid, the returned value should be None, then in line 35, the function add a key value pair to db.

Yes the value need to be JSON, this will be checked in the functions where the data will be checked. Here in the db_utils, it'll just insert and read whatever is given. I agree we'll need some tests for DbUtils class...

Reply:
Created https://github.com/atlanticwave-sdx/sdx-controller/issues/131 to track tests for DbUtils.